src/controllers/connection/index.ts

```diff
@@ -118,4 +152,52 @@ export class ConnectionController extends HTMLController {
       ],
     })
   }
+  private async pollPinSubmission(connectionId: string, initialPinAttemptsRemaining: number) {
+    const startTime = Date.now()
+    const timeout = 4000 // 4 seconds
+    const interval = 100 // 100 ms
+
+    while (Date.now() - startTime < timeout) {
+      const [connectionCheck]: ConnectionRow[] = await this.db.get('connection', { id: connectionId })
+      this.logger.debug('Polling for local pin attempts change.')
+      //first check if we have exceeded the number of pin attempts -> redirect to connections page
+      if (initialPinAttemptsRemaining > 0 && connectionCheck.pin_tries_remaining_count == 0) {
```

Comment:
why 2 separate counters for the pin attempts/tries remaining?

Reply:
because there is a different number of tries remaining before you submit the pin and after you do so/you receive a problem report